internal/resources/provisioner/data_source_provisioner.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright © 2021 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package provisioner
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/authctx"
+	clienterrors "github.com/vmware/terraform-provider-tanzu-mission-control/internal/client/errors"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/helper"
+	provisioner "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/provisioner"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/common"
+)
+
+func DataSourceProvisioner() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: func(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+			return dataSourceProvisionerRead(helper.GetContextWithCaller(ctx, helper.DataRead), d, m)
+		},
+		Schema: provisionerSchema,
+	}
+}
+
+func dataSourceProvisionerRead(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	config := m.(authctx.TanzuContext)
+
+	provisionerName, ok := d.Get(NameKey).(string)
+	if !ok {
+		return diag.Errorf("unable to read provisioner name")
+	}
+
+	managementClusterName, ok := d.Get(ManagementClusterNameKey).(string)
+	if !ok {
+		return diag.Errorf("unable to read management cluster name")
+	}
+
+	fn := &provisioner.VmwareTanzuManageV1alpha1ManagementclusterProvisionerFullName{
+		Name:                  provisionerName,
+		ManagementClusterName: managementClusterName,
+	}
+
+	resp, err := config.TMCConnection.ProvisionerResourceService.ProvisionerResourceServiceGet(fn)
```

Comment:
The purpose of adding the datasource is that user can see all the existing provisioners in a management cluster, this Get API would just return the info of the specified provisioner and not list all the provisioners. So this should be replaced with List call and not GET, agree?

Reply:
Agree, add the data source support to list all the provisioners in a management cluster as well as to read individual provisioner also